src/prefect/client/orchestration.py

```diff
@@ -2222,10 +2226,11 @@ async def create_task_run(
             state=state.to_state_create(),
             task_inputs=task_inputs or {},
         )
+        json = task_run_data.dict(json_compatible=True)
```

Comment:
The `json_compatible` flag is quite slow and I've been replacing it ahead of Pydantic v2 cutover (as it serializes and then reloads the json) - I'd recommend something like this for efficiency:

```
data = task_run_data.json(exclude={'id'} if id is None else None)
response = self._client.post('/task_runs/', data=data)
```